javascript/ql/lib/change-notes/2024-11-28-regexp-unknown-flags.md

```diff
@@ -0,0 +1,5 @@
+---
+category: majorAnalysis
+---
+* Queries working with regular expressions now also handle regular expressions constructed using `new RegExp(..)`. Previously some queries didn't handle such regular expressions. 
```

Comment:
```suggestion
* The `js/incomplete-sanitization` query now also checks regular expressions constructed using `new RegExp(..)`. Previously it only checked regular expression literals.
* Regular expression-based sanitisers implemented with `new RegExp(..)` are now detected in more cases.
```
I'd try to split this up to be more accurate. We already have quite good coverage for `new RegExp(..)` and IMO the change note made it sound like things were in a worse state than they were.